pool/object.go

```diff
@@ -40,104 +40,182 @@ func (p *Pool) ObjectPutInit(ctx context.Context, hdr object.Object, prm client.
+	start := time.Now()
+	reader, err := c.ObjectRangeInit(ctx, containerID, objectID, offset, length, prm)
+	statUpdater.incRequests(time.Since(start), methodObjectRange)
+	statUpdater.updateErrorRate(err)
+
+	return reader, err
 }
 
 // ObjectDelete marks an object for deletion from the container using NeoFS API protocol.
 //
 // See details in [client.Client.ObjectDelete].
 func (p *Pool) ObjectDelete(ctx context.Context, containerID cid.ID, objectID oid.ID, prm client.PrmObjectDelete) (oid.ID, error) {
-	c, err := p.sdkClient()
+	c, statUpdater, err := p.sdkClient()
 	if err != nil {
 		return oid.ID{}, err
 	}
-
-	if err = p.withinContainerSession(ctx, c, containerID, p.actualSigner(prm.Signer()), session.VerbObjectDelete, &prm); err != nil {
+	if err = p.withinContainerSession(
+		ctx,
+		c,
+		containerID,
+		p.actualSigner(prm.Signer()),
+		session.VerbObjectDelete,
+		&prm,
+		statUpdater,
+	); err != nil {
 		return oid.ID{}, fmt.Errorf("session: %w", err)
 	}
 
-	return c.ObjectDelete(ctx, containerID, objectID, prm)
+	start := time.Now()
+	id, err := c.ObjectDelete(ctx, containerID, objectID, prm)
+	statUpdater.incRequests(time.Since(start), methodObjectDelete)
+	statUpdater.updateErrorRate(err)
+
+	return id, err
 }
 
 // ObjectHash requests checksum of the range list of the object payload using
 //
 // See details in [client.Client.ObjectHash].
 func (p *Pool) ObjectHash(ctx context.Context, containerID cid.ID, objectID oid.ID, prm client.PrmObjectHash) ([][]byte, error) {
-	c, err := p.sdkClient()
+	c, statUpdater, err := p.sdkClient()
 	if err != nil {
 		return [][]byte{}, err
 	}
-
-	if err = p.withinContainerSession(ctx, c, containerID, p.actualSigner(prm.Signer()), session.VerbObjectRangeHash, &prm); err != nil {
+	if err = p.withinContainerSession(
+		ctx,
+		c,
+		containerID,
+		p.actualSigner(prm.Signer()),
+		session.VerbObjectRangeHash,
+		&prm,
+		statUpdater,
+	); err != nil {
 		return [][]byte{}, fmt.Errorf("session: %w", err)
 	}
 
-	return c.ObjectHash(ctx, containerID, objectID, prm)
+	hashes, err := c.ObjectHash(ctx, containerID, objectID, prm)
+	statUpdater.updateErrorRate(err)
+
+	return hashes, err
 }
 
 // ObjectSearchInit initiates object selection through a remote server using NeoFS API protocol.
 //
 // See details in [client.Client.ObjectSearchInit].
 func (p *Pool) ObjectSearchInit(ctx context.Context, containerID cid.ID, prm client.PrmObjectSearch) (*client.ObjectListReader, error) {
-	c, err := p.sdkClient()
+	c, statUpdater, err := p.sdkClient()
 	if err != nil {
 		return nil, err
 	}
-
-	if err = p.withinContainerSession(ctx, c, containerID, p.actualSigner(prm.Signer()), session.VerbObjectSearch, &prm); err != nil {
+	if err = p.withinContainerSession(
+		ctx,
+		c,
+		containerID,
+		p.actualSigner(prm.Signer()),
+		session.VerbObjectSearch,
+		&prm,
+		statUpdater,
+	); err != nil {
 		return nil, fmt.Errorf("session: %w", err)
 	}
```

Comment:
`statUpdater.updateErrorRate(err)` here?

Reply:
And a counter as well.

Reply:
In this method we pass `statUpdater` inside and it counts error rate inside, when call the network. This `updateErrorRate` calculates only network errors. Inside `withinContainerSession` may be many different errors

Reply:
`ObjectSearch` was noone, it is fixed now. I though You mean put counter inside 219 line

Reply:
I wasn't able to find a better line for this comment!